azkaban-web-server/src/main/java/azkaban/flowtrigger/plugin/FlowTriggerDependencyPluginManager.java

```diff
@@ -108,35 +122,42 @@ private DependencyPluginConfig mergePluginConfig(final Map<String, String> publi
     return new DependencyPluginConfigImpl(combined);
   }
 
-
   private DependencyCheck createDependencyCheck(final DependencyPluginConfig pluginConfig)
       throws FlowTriggerDependencyPluginException {
     final String classPath = pluginConfig.get(CLASS_PATH);
+
     final String[] cpList = classPath.split(",");
 
     final List<URL> resources = new ArrayList<>();
 
     try {
       for (final String cp : cpList) {
-        final URL cpItem = new File(cp).toURI().toURL();
-        if (!resources.contains(cpItem)) {
-          logger.info("adding to classpath " + cpItem);
-          resources.add(cpItem);
+        final File[] files = getFilesMatchingPath(cp);
+        if (files != null) {
+          for (final File file : files) {
+            final URL cpItem = file.toURI().toURL();
+            if (!resources.contains(cpItem)) {
+              logger.info("adding to classpath " + cpItem);
+              resources.add(cpItem);
+            }
+          }
         }
       }
     } catch (final Exception ex) {
       throw new FlowTriggerDependencyPluginException(ex);
     }
 
-    final ClassLoader dependencyClassloader =
-        new URLClassLoader(resources.toArray(new URL[resources.size()]));
+    final ClassLoader dependencyClassloader = new ParentLastURLClassLoader(
+        resources.toArray(new URL[resources.size()]), this.getClass().getClassLoader());
+
+    Thread.currentThread().setContextClassLoader(dependencyClassloader);
+
     Class<? extends DependencyCheck> clazz = null;
     try {
       clazz = (Class<? extends DependencyCheck>) dependencyClassloader.loadClass(pluginConfig.get
           (DEPENDENCY_CLASS));
-      final DependencyCheck dependencyCheck =
-          (DependencyCheck) Utils.callConstructor(clazz);
-      return dependencyCheck;
+      final Object obj = Utils.callConstructor(clazz);
```

Comment:
Is there a reason we're using `Object` here instead of a closer shared parent class? Does `DependencyCheck` fail to work now? I might not understand.

Also, even if we're using `Object` we can probably still use the same more descriptive name of `dependencyCheck` for the variable.

Reply:
thanks, updated.